lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -0,0 +1,243 @@
+  LogicalResult
+  matchAndRewrite(AtenBincountOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op.getLoc();
+    Value input = adaptor.self();
+    Value minlength = adaptor.minlength();
+    Value weights = adaptor.weights();
+
+    // Check whether the input is a 1-d tensor or not.
+    RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+    if (inputType.getRank() != 1)
+      return rewriter.notifyMatchFailure(
+          op, "Input tensor has to be one dimensional.");
+
+    // TODO: Incorporate the weight argument.
+    if (!weights.getType().isa<mlir::torch::Torch::NoneType>())
+      return rewriter.notifyMatchFailure(
+          op, "Unimplemented, the weights operand is not incorporated.");
+
+    // Finding the maximum value in the input tensor.
+    SmallVector<int64_t> maxTensorSizes;
+    ValueTensorType maxTensorType = ValueTensorType::get(
+        op->getContext(), llvm::makeArrayRef(maxTensorSizes),
+        op.self().getType().cast<ValueTensorType>().getDtype());
+    Value maxTensor = rewriter.create<AtenMaxOp>(loc, maxTensorType, op.self());
+    TypeConverter *typeConverter = getTypeConverter();
+    maxTensor = typeConverter->materializeTargetConversion(
+        rewriter, loc, typeConverter->convertType(maxTensor.getType()),
+        maxTensor);
+
+    // `maxTensor` is a 0-d tensor, extracting its only element and
+    // storing it in `maxInput`.
+    Value maxInput = rewriter.create<tensor::ExtractOp>(loc, maxTensor);
+
+    // Creating a tm_tensor.scatter op with the following mapping:
+    // 1.) `input` tensor maps to the indices in scatter op. `input` is
+    // expanded from 1-d to 2-d, and its element type is set to i32 as required
+    // for the scatter op.
+    // 2.) `updates` is a 1-d dummy tensor with the size equivalent to the
+    // `input`.
+    // 3.) `bincount` a 1-d tensor maps to the original in scatter op
+    // with size equal to the max(max(input) + 1, minlength).
+    SmallVector<int64_t> expandedInputSizes{inputType.getShape()[0], 1};
+    ValueTensorType expandInputType = ValueTensorType::get(
+        op->getContext(), llvm::makeArrayRef(expandedInputSizes),
+        op.self().getType().cast<ValueTensorType>().getDtype());
+    Value torchCstOne = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(1));
+    Value expandedInputTensor = rewriter.create<AtenUnsqueezeOp>(
+        loc, expandInputType, op.self(), torchCstOne);
+
+    // Converting the input element type from i64 to i32.
+    ValueTensorType truncatedTensorType = ValueTensorType::get(
+        op->getContext(), llvm::makeArrayRef(expandedInputSizes),
+        mlir::IntegerType::get(op->getContext(), 32,
+                               mlir::IntegerType::Signed));
+    int intType = (int)ScalarType::Int;
```

Comment:
there is a helper for type conversion in https://github.com/llvm/torch-mlir/blob/486f95e84f587d020ba789b071b12f890510f1a1/lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp#L130. Can you move the helper to Torch/utils/utils.cpp and use the helper here.